data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -1,165 +1,130 @@
 """)
 
-# Deactivated participant table fields to query off of
-PID_TABLE_FIELDS = [[{
-    "type": "integer",
-    "name": "person_id",
-    "mode": "required",
-    "description": "The person_id to retract data for"
-}, {
-    "type": "date",
-    "name": "deactivated_date",
-    "mode": "required",
-    "description": "The deactivation date to base retractions on"
-}]]
-
-
-def get_pids_table_info(project_id, dataset_id, client):
-    """
-    This function gets all table information for the dataset and filters to only retain tables with the field person_id
 
-    :param project_id: bq name of project_id
-    :param dataset_id: ba name of dataset_id
-    :param client: bq client object
-    :return: dataframe with table information schema from BigQuery
+def get_date_cols_dict(date_cols_list):
     """
-    all_table_info_query = TABLE_INFORMATION_SCHEMA.render(project=project_id,
-                                                           dataset=dataset_id)
-    result_df = client.query(all_table_info_query).to_dataframe()
-    # Get list of tables that contain person_id
-    pids_tables = []
-    for _, row in result_df.iterrows():
-        column = getattr(row, 'column_name')
-        table = getattr(row, 'table_name')
-        if 'person_id' in column:
-            pids_tables.append(table)
-
-    pids_table_info_df = pd.DataFrame()
-
-    if pids_tables and len(result_df) != 0:
-        pids_table_info_df = result_df[result_df['table_name'].isin(
-            pids_tables)]
-
-    return pids_table_info_df
+    Converts list of date/datetime columns into dictionary mappings
 
-
-def get_table_dates_info(project_id, dataset_id, client):
+    Assumes each date column has a corresponding datetime column due to OMOP specifications
+    If a date column does not have a corresponding datetime column, skips it
+    :param date_cols_list: list of date/datetime columns
+    :return: dictionary with mappings for START_DATE, START_DATETIME, END_DATE, END_DATETIME
+        or DATE, DATETIME
+    """
+    date_cols_dict = {}
+    for field in date_cols_list:
+        if field.endswith(consts.START_DATETIME):
+            date_cols_dict[consts.START_DATETIME] = field
+        elif field.endswith(consts.END_DATETIME):
+            date_cols_dict[consts.END_DATETIME] = field
+        elif field.endswith(consts.DATETIME):
+            date_cols_dict[consts.DATETIME] = field
+    for field in date_cols_list:
+        if field.endswith(consts.START_DATE):
+            if date_cols_dict.get(consts.START_DATETIME, '').startswith(field):
+                date_cols_dict[consts.START_DATE] = field
+        elif field.endswith(consts.END_DATE):
+            if date_cols_dict.get(consts.END_DATETIME, '').startswith(field):
+                date_cols_dict[consts.END_DATE] = field
+        elif field.endswith(consts.DATE):
+            if date_cols_dict.get(consts.DATETIME, '').startswith(field):
+                date_cols_dict[consts.DATE] = field
```

Comment:
I tried to find all instances of column names having `date` in the name. Almost all of them had a corresponding date/datetime column thanks to OMOP except for `birth_datetime` in `person` and `verbatim_end_date` in `drug_exposure`. The person table is excluded since we are retracting only EHR data, but I wasn't sure if I had to include `verbatim_end_date`. Let me know your thoughts. `visit_date` doesn't exist and I used it as an example to exclude such dates.

Reply:
My initial thought is to ask, was `verbatim_end_date` specifically excluded in the original design?  If it was excluded, this should work fine.  If it was included, you can exclude the person table.

Reply:
I looked through the old tickets and inclusion/exclusion does not seem to be mentioned wrt any dates. I'll add this in.

Reply:
So,did anything change here?  It seems like this entire function can be simplified to:
```
    date_cols_dict = {}
    for field in date_cols_list:
        if field.endswith(consts.START_DATETIME):
            date_cols_dict[consts.START_DATETIME] = field
        elif field.endswith(consts.END_DATETIME):
            date_cols_dict[consts.END_DATETIME] = field
        elif field.endswith(consts.DATETIME):
            date_cols_dict[consts.DATETIME] = field
        elif field.endswith(consts.START_DATE):
            date_cols_dict[consts.START_DATE] = field
        elif field.endswith(consts.END_DATE):
            date_cols_dict[consts.END_DATE] = field
        elif field.endswith(consts.DATE):
            date_cols_dict[consts.DATE] = field
```

I don't think you even need to exclude the `person` table here because the only column that fits the `datetime` naming structure is `birth_datetime` and it should be logically impossible for someone to deactivate their account before their birth.

Reply:
The person table columns won't be passed to this function because it is excluded before it gets to this stage. 

The problem with this simplification is some dates can mess up the `COALESCE` hierarchy in the query. E.g. if `verbatim_end_date` is encountered before `drug_exposure_end_date`, it will be used in the hierarchy instead of the intended order of precedence.

Reply:
Ok.  This is helping.  What precedence is assumed?  There is no mention of this in the docstring for this function or the comments.  Since this is implementing some business logic, please add some clarifying comments on this.

Reply:
I'll add this in the docstring